java/client/src/org/openqa/selenium/support/pagefactory/ByChained.java

```diff
@@ -61,16 +61,14 @@ public WebElement findElement(SearchContext context) {
       return new ArrayList<>();
     }
 
-    List<WebElement> elems = null;
-    for (By by : bys) {
+    List<WebElement> elems = bys[0].findElements(context);
+    for (int i = 1; i < bys.length; i++) {
+      if (elems.isEmpty()) {
```

Comment:
Isn't this changing the previous behavior? 
I see in the current code that even if `elems` is empty, the loop continues through the rest of the `bys` array.

Reply:
Hey @diemol, thanks for taking the time to look into this.

I understand your point. It looks like it behaves that way, but actually it doesn't. This is exactly the point of this PR.
BTW, the same concern was also pointed out in the other review (https://github.com/SeleniumHQ/selenium/pull/372#discussion_r27221662).

I purposefully added two unit tests as the first commit, *before* committing the refactoring.
As I wrote in the commit message:
> add unit tests to verify the assumption that in case the chain is broken,
> i.e. one of the bys in the chain yields no elements,
> then ByChained#findElements returns an empty list.
 
You may checkout 51c5bd690cb7783560d1accddae263d44e26bc4a and see they are passing.

As I pointed out in https://github.com/SeleniumHQ/selenium/pull/372#discussion_r34129844:

> The key is `elems = newElems;` at the end of each iteration, which overwrites the list that is iterated during each step. The first by that returns no elements will set elems to empty list; any subsequent iteration of the outer loop will do nothing, since the inner loop will not be executed.

Reply:
Thanks for clarifying.